pkg/edgeview/dispatcher/wss-server.go

```diff
@@ -0,0 +1,265 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type endPoint struct {
+	hostname  string
+	wsConn    *websocket.Conn
+}
+
+const (
+	noDeviceMsg string = "no device online\n+++Done+++"
+	tokenReqMsg string = "token is required"
+	moretwoMsg  string = "can't have more than 2 peers"
+)
+
+var upgrader = websocket.Upgrader{} // use default options
+// reqAddrTokeConn indexed by 'toke' then 'remoteAddr' strings
+var reqAddrTokenEP   map[string]map[string]endPoint
+// mutex for access the maps
+var connMutex sync.Mutex
+// connection id, keep inc
+var connID int
+// debug set
+var needDebug bool
+
+func socketHandler(w http.ResponseWriter, r *http.Request) {
+    // Upgrade our raw HTTP connection to a websocket based one
+    conn, err := upgrader.Upgrade(w, r, nil)
+    if err != nil {
+        fmt.Printf("Error during connection upgradation: %v\n", err)
+        return
+    }
+    defer conn.Close()
+
+	if _, ok := r.Header["X-Session-Token"]; !ok {
+		err := conn.WriteMessage(websocket.TextMessage, []byte(tokenReqMsg))
+		if needDebug {
+			fmt.Printf("websocket write: %v\n", err)
+		}
+		return
+	}
+	if len(r.Header["X-Session-Token"]) == 0 {
+		err := conn.WriteMessage(websocket.TextMessage, []byte(tokenReqMsg))
+		if needDebug {
+			fmt.Printf("websocket write: %v\n", err)
+		}
+		return
+	}
+	token := r.Header["X-Session-Token"][0]
```

Comment:
Is this token the binding factor between edge-view client and edge-node?

Reply:
yes, the token is the only thing to bind the sessions.